src/isolate_setup.cpp

```diff
@@ -63,8 +76,8 @@ Dart_Isolate CreateKernelIsolate(const char* script_uri,
   intptr_t kernel_service_buffer_size = 0;
   dfe.LoadKernelService(&kernel_service_buffer, &kernel_service_buffer_size);
 
-  IsolateGroupData* isolate_group_data =
-      new IsolateGroupData(uri, package_config, nullptr, false);
+  DllIsolateGroupData* isolate_group_data =
```

Comment:
In cases of non-user isolates (the Kernel and Service isolates) I think it's better to use the unadorned IsolateGroupData. Only for the user isolates does it makes sense to have this new version.

Reply:
Since the cast is a static_cast (i.e. no type safety) i was thinking its better to be safe encase a user calls it with a ptr got from any isolate.

Reply:
We could (maybe, I forget the compiler flags) switch to a dynamic cast. But, also, we shouldn't ever send the Kernal or Service isolate handles back to the user, so there's very little chance they'll ask for the group data from them.